soccer/planning/Trajectory.hpp

```diff
@@ -0,0 +1,422 @@
+#pragma once
+
+#include <Geometry2d/Pose.hpp>
+#include <time.hpp>
+
+#include "DebugDrawer.hpp"
+#include "Instant.hpp"
+#include "Utils.hpp"
+#include "planning/DynamicObstacle.hpp"
+
+namespace Planning {
+
+/**
+ * @brief An ordered sequence (by time) of RobotInstants
+ */
+using RobotInstantSequence = std::vector<RobotInstant>;
+
+/**
+ * @brief Represents a trajectory x(t), y(t), h(t), with a smooth velocity and
+ * piecewise-continuous acceleration.
+ *
+ * @details Note that this provides two interfaces: one accepting RJ::Seconds,
+ * which is a duration, and one accepting RJ::Time, an instant in time.
+ *
+ * This uses cubic interpolation between two adjacent instants, fitting a
+ * parametric cubic polynomial based on initial and final posiiton and velocity.
+ *
+ * A trajectory also carries a timestamp representing the time at which it was
+ * planned. This should be filled in using `trajectory.stamp(RJ::Time)` before
+ * publishing, and will be reset to `nullopt` when the trajectory is modified.
+ */
+class Trajectory {
+public:
+    Trajectory() = default;
+    ~Trajectory() = default;
+
+    /**
+     * @brief Create a trajectory from several "instants", each with a pose,
+     * velocity, and timestamp. This
+     */
+    explicit Trajectory(RobotInstantSequence instants)
+        : instants_(std::move(instants)) {}
+
+    /**
+     * @brief Create a trajectory from two other trajectories.
+     *  Assumes a.last() == b.first() so b.first() is skipped.
+     *
+     * @details Is able to move out of the first trajectory, but not the second.
+     */
+    Trajectory(Trajectory a, const Trajectory& b);
```

Comment:
Maybe have an r-value reference / smart pointer version of this function, so we can move out of the first Trajectory. Here we are copying both.

Reply:
See above